### ISSUE TITLE
Run officer table script to fix roster issues

### ISSUE THREAD

**Issue by claireholt-tw** [PublicDataWorks/complaint-manager#477]
### Business Context:
We need to run a script to clean up duplicated officers, so NOIPM can upload their updated roster table, and it can behave as expected. 

### In Scope
1. Download officers table and cases_officers table as CSVs

2. Read data from each CSV. Set up two output files for our final pruned tables. This resulting officer table will no longer have duplicates (except ones we manually adjust, i.e. differing names on same badge number).

3. Step through officers table iteratively. On each iteration, check if the badge number matches any badge number we have in our list of previously sorted officers (an array just containing badge numbers). If they aren't in the list, continue & add their badge number to the list of previously sorted officers. Otherwise, skip them. 

4. Iterate over the entirety of our original officers CSV and search for duplicate badge numbers (this will identify duplicate officers)

5. Add these duplicates to an array (to preserve the duplicate officer instances to be consolidated later)

6. Check to see if names of duplicate officers from step above match. If names are different but badge number is the same, we will need to address that manually later, for now, immediately adding them as a distinct officer following protocol in step 7.
Question: Are badge numbers unique? What if two people have the same badge number?
Note: `windows_username` in the `officers` table is equivalent to Badge ID (as NO calls it)

7. Compare duplicate officers and find officer that has been most recently updated (this will be our consolidated officer that will persist in new table)
** If an officer isn't duplicated, add them as a new row in the officers CSV and search thru cases_officers for any mention of them and and those rows to the new cases_officers CSV.

8. Once finding the most recently updated officer from the step above, add them to the new officers CSV.

9. Iterate through cases_officers in search of the IDs of every single duplicate we found - Look thru cases_officer for any officer_id that match any of our duplicated officers id field and create a copy of each of these rows.

10. Alter the officer_id from step above to match the id of our consolidated officer

11. Add each of these new rows to our new cases_officers CSV.

12. The above steps repeat until we finish iterating thru the officers table.

Once the script is done, check the outputs to make sure it looks right.

Duplicate the officers table and cases_officers table in prod so we can test our scripts there before messing with current tables (renamed as test tables or something different)

Import new CSV data into these tables starting with officers then cases_officers. Troubleshoot :)

If all looks good here, brace yourself and drop tables in prod and add new ones in the way you did previously using the import CSV option in DBeaver.

** Shak may need to reach out to Christian to get access to drop tables and potentially other actions and be on call when we update the real tables so he can test it.

### Out of Scope
1. 

### Notes
KEEP AN UNALTERED CSV OF ORIGINAL DATA, never mutate this source
When using prod data in local to test, make sure to delete all instances locally at the end of the story.
The CSV of the final roster that Marilia made will have to be checked/modified to match the new outputted CSV. [Discussion with @rnair-tw and @sbhamani

**Comment by rnair-tw** [comment]
To split story
- Get the script working in duplicate tables, generate the CSVs
- Get this working in prod, update the final table w/ the new roster

**Comment by jeanwatts** [comment]
How to ensure data looks good:
-all officers have distinct windows_usernames
-all cases_officers data is preserved and officer_id field corresponds to an id in officers table who seems like the correct officer
-unique officers with quality data were not lost (some officers with little to know info besides their names were intentionally thrown out)

**Comment by claireholt-tw** [comment]
Notes from desk check:

1. Make a card renaming windows_username to be employee_id, make this field mandatory and force it to be unique

**Comment by thoughtworks-vwong** [comment]
Verified:

- All officers have distinct windows_usernames
- All cases_officers data is preserved and officer_id field corresponds to an id in officers table who seems like the correct officer
- Unique officers with quality data were not lost (some officers with little to know info besides their names were intentionally thrown out)

Formulas Used in Excel:

=COUNTIF ($S$2:$S$5733, S2)>1
=COUNTIF(officers_table_noipm!A:A,C2)>0

**Comment by rnair-tw** [comment]
Script was run
